Jenkinsfile.trigger

```
@@ -17,6 +17,35 @@ def breakEarly = false // thanks Jenkins...
 // { buildId: { "count": 1, ... }, ... }
 def pastFailedJobsJson = '{}'
 
+// globals: (so they can be accessed in the setFailedJob function)
+// pastFailedJobsJson as a native object
+pastFailedJobs = [:]
+// object to fill failed jobs json after queue completion, includes skipped
+newFailedJobs = [:]
+
+def setFailedJob(buildId, identifier, timeNow, url) {
+	if (pastFailedJobs[buildId]) {
+		// carry forward count(GHA triggers)/time
+		newFailedJobs[buildId] = [
+			count: pastFailedJobs[buildId].count + 1,
+			firstTime: pastFailedJobs[buildId].firstTime,
+			identifier: identifier,
+			lastTime: timeNow,
+			skips: 0,
+			url: url,
+		]
+	} else {
+		newFailedJobs[buildId] = [
+			count: 1,
+			firstTime: timeNow,
+			identifier: identifier,
+			lastTime: timeNow,
+			skips: 0,
+			url: url,
+		]
+	}
```

If we pull out the "else" block, the common bits of this can collapse (and then we have a more obvious place to "fix" `firstTime` for existing entries later, if we need to:

```suggestion
	newFailedJobs[buildId] = [
		count: 1,
		firstTime: timeNow,
		identifier: identifier,
		lastTime: timeNow,
		skips: 0,
		url: url,
	]
	if (pastFailedJobs[buildId]) {
		// carry forward count (fails/GHA triggers) + time
		newFailedJobs[buildId].count += pastFailedJobs[buildId].count
		newFailedJobs[buildId].firstTime = pastFailedJobs[buildId].firstTime
	}
```

(unfortunately we can't use `Math.min`/`Math.max` to get `firstTime` fixing for "free" because `null` isn't a number, but we could maybe do something clever with the "elvis operator"; `1 ?: 2` => 1, but `null ?: 2` => 2)

---

Maybe something like `newFailedJobs[buildId].firstTime = Math.min(pastFailedJobs[buildId].firstTime ?: Long.MAX_VALUE, newFailedJobs[buildId].firstTime)` ? (also totally happy to punt on this - we're not using the value yet, so it's harmless for now and we can keep the easier to read value and we might not even have any "old" data left by the time we want to use it for something)